### ISSUE TITLE
E6101 {'Fn::GetAtt': ['HostedZone', 'NameServers']} is not of type 'array'

### ISSUE THREAD

**Issue by jakewaggoner** [aws-cloudformation/cfn-lint#3648]
### CloudFormation Lint Version

1.12.0

### What operating system are you using?

Mac

### Describe the bug

I have an output on a CFT that look like this:
```
Outputs:
  HostedZoneNameServers:
    Value: !Join
      - ","
      - !GetAtt HostedZone.NameServers
```

This was working fine in version < 1.12.0, however now I get the error saying `E6101 {'Fn::GetAtt': ['HostedZone', 'NameServers']} is not of type 'array'`.

The `HostedZone` resource is of type `AWS::Route53::HostedZone`.

### Expected behavior

The output should take the array of name servers and concatenate them with a comma, producing a CSV string.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  HostedZone:
    Type: AWS::Route53::HostedZone
    Properties:
      Name: 'myexamplesite.com'

Outputs:
  HostedZoneNameServers:
    Value: !Join
      - ","
      - !GetAtt HostedZone.NameServers
```

**Comment by kddejong** [comment]
@jakewaggoner Looking into it. Just went through the opposite last week so doing some quick tests.

**Comment by kddejong** [comment]
Okay related to #3635.  Looks like while both are arrays of strings (in the resource schema) that resource is actually returning a string.  I thought that was a GetAtt thing but I'm now thinking that is just wrongly defined in the schema.  

Going to dig a little deeper and see if I can validate my exception.

**Comment by kddejong** [comment]
Undoing the change and patching the `AWS::ElastiCache::ReplicationGroup` resource to have the correct type.  Should have a fix in shortly with a patch release shortly after that.

**Comment by jakewaggoner** [comment]
> Undoing the change and patching the `AWS::ElastiCache::ReplicationGroup` resource to have the correct type. Should have a fix in shortly with a patch release shortly after that.

The speed at which you fix things is truly incredible. Thanks for the work you do!
